packages/@o3r/core/schematics/library/rules/shared.ts

```diff
@@ -0,0 +1,88 @@
+import * as path from 'node:path';
+import type { Rule } from '@angular-devkit/schematics';
+import type { PackageJson } from 'type-fest';
+import { NgGenerateModuleSchema } from '../schema';
+import { getPackageManagerRunner, getWorkspaceConfig } from '@o3r/schematics';
+
+/**
+ * Generate rule to update generated package.json file
+ *
+ * @param targetPath Path of the generated files
+ * @param otterVersion Current version of otter
+ * @param o3rCorePackageJson Content of core's package.json
+ * @param options Option of the schematic
+ */
+export function updatePackageDependenciesFactory(
+    targetPath: string,
+    otterVersion: string,
+    o3rCorePackageJson: PackageJson & { generatorDependencies?: Record<string, string>},
+    options: NgGenerateModuleSchema): Rule {
+  return (tree) => {
+    /* eslint-disable @typescript-eslint/naming-convention */
+    const packageJson = tree.readJson(path.posix.join(targetPath, 'package.json')) as PackageJson;
+    const runner = getPackageManagerRunner(getWorkspaceConfig(tree));
+    packageJson.description = options.description || packageJson.description;
+    packageJson.scripts ||= {};
+    packageJson.scripts.build = `${runner} build ${options.projectName}`;
+    packageJson.scripts['prepare:build:builders'] = `${runner} cpy 'collection.json' dist && ${runner} cpy 'schematics/**/*.json' dist/schematics`;
+    packageJson.scripts['build:builders'] = 'tsc -b tsconfig.builders.json --pretty';
+    packageJson.peerDependencies ||= {};
+    packageJson.peerDependencies['@o3r/dev-tools'] = otterVersion;
+    packageJson.peerDependencies['@o3r/schematics'] = otterVersion;
+    packageJson.devDependencies ||= {};
+    packageJson.keywords ||= [];
+    packageJson.keywords.push('otter-module');
+    packageJson.version = '0.0.0-placeholder';
+
+    packageJson.devDependencies = {
+      ...packageJson.devDependencies,
+      '@angular-devkit/build-angular': o3rCorePackageJson.peerDependencies!['@angular-devkit/core'],
+      '@angular-devkit/core': o3rCorePackageJson.peerDependencies!['@angular-devkit/core'],
+      '@angular-eslint/eslint-plugin': o3rCorePackageJson.generatorDependencies!['@angular-eslint/eslint-plugin'],
+      '@angular/cli': packageJson.peerDependencies['@angular/common'],
+      '@angular/common': packageJson.peerDependencies['@angular/common'],
+      '@angular/compiler': packageJson.peerDependencies['@angular/common'],
+      '@angular/compiler-cli': packageJson.peerDependencies['@angular/common'],
+      '@angular/core': packageJson.peerDependencies['@angular/common'],
+      '@angular/platform-browser': packageJson.peerDependencies['@angular/common'],
+      '@angular/platform-browser-dynamic': packageJson.peerDependencies['@angular/common'],
+      '@o3r/dev-tools': otterVersion,
+      '@o3r/eslint-plugin': otterVersion,
+      '@schematics/angular': o3rCorePackageJson.peerDependencies!['@angular-devkit/core'],
```

Comment:
```suggestion
      '@schematics/angular': o3rCorePackageJson.peerDependencies!['@schematics/angular'],
```

or we don't have this peerdep? Anyway, I am not sure `@schematics` and `@angular-devkit` are aligned.